File: tools/apigen/use_fidget.js
fidget = require("./fidget")

// Test fidget
var count = 0
clickCb = fidget.cb(function() {
    count += 1
    console.log("count:", count)
})
fidget.addCb(fidget.E_ON_CLICK, 100, "/CounterFrame/Count1Up", clickCb)

displayCb = fidget.cb(function() {
    n = fidget.findNode("text")
    n.characters = count.toString()
    n.dirty = true
})
fidget.addCb(fidget.E_ON_DISPLAY, 100, "/CounterFrame/CounterDisplay", displayCb)

fidget.startFidget(
  "https://www.figma.com/file/Km8Hvdw4wZwEk6L1bN4RLa",
  "JavaScript Counter",
  "CounterFrame",
  false
)

File: tools/apigen/use_fidget.nim
import fidgetapi

# Test fidget
var count = 0
proc clickCb() {.cdecl.} =
  count += 1
  echo("count: ", count)
addCb(eOnClick, 100, "/CounterFrame/Count1Up", clickCb)

proc display_cb() {.cdecl.} =
  var n = find_node("text")
  n.characters = $count
  n.dirty = true
addCb(eOnDisplay, 100, "/CounterFrame/CounterDisplay", display_cb)

startFidget(
  "https://www.figma.com/file/Km8Hvdw4wZwEk6L1bN4RLa",
  "Nim Counter",
  "CounterFrame",
  false
)
